src/_pytest/nodes.py

```diff
@@ -247,14 +248,18 @@ def setup(self) -> None:
     def teardown(self) -> None:
         pass
 
-    def listchain(self) -> List["Node"]:
-        """ return list of all parent collectors up to self,
-            starting from root of collection tree. """
-        chain = []
+    def iterchain(self) -> Generator["Node", None, None]:
```

Comment:
```suggestion
    def iterchain(self) -> Iterator["Node"]:
```

Reply:
curious about this suggestion, Generator is the more correct type here

Reply:
I prefer `Iterator` here because it gives us more leeway to implement it however we want. If we specify a `Generator` that locks us to a `yield` implementation, and we don't gain anything in return (we really don't want someone to start `send`ing or `throw`ing into the yield here...)

Reply:
Interesting, this clarifies to me the difference now, thanks. 👍